UnityProject/Assets/Scripts/Util/SweetExtensions.cs

```diff
@@ -166,6 +166,19 @@ public static GameObject GetRootGameObject(this GameObject go)
 		}
 	}
 
+	/// Creates garbage! Use very sparsely!
+	public static CommonComponents GetCommonComponents(this GameObject go)
+	{
+		if (ComponentManager.TryGetCommonComponent(go, out  var commonComponent))
+		{
+			return commonComponent;
+		}
+		else
+		{
+			return null;
+		}
```

Comment:
```cs
if (ComponentManager.TryGetCommonComponent(go, out var commonComponent) == false) return null;
return commonComponent;
```

Reply:
``return ComponentManager.GetComponent<CommonComponent>()`` would it be the same?

Reply:
GetComponent fails unexpectedly sometimes, use TryGetComponent because its safer

Reply:
makes it harder to read if compacting it